src/release_notes_workflow/component_release_notes.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+
+from manifests.input_manifest import InputComponentFromSource
+
+
+class ComponentReleaseNotes:
+
+    def __init__(self, component: InputComponentFromSource, build_version: str, root: str) -> None:
+        self.component = component
+        self.build_version = build_version
+        self.root = root
+
+    @property
+    def filename(self) -> str:
```

Comment:
Sure I have added as 
```
    @property
    @abstractmethod
    def filename(self) -> str:
        pass
 ```
 
 So that the function `def exists(self)` can still infer as property.
 @dblock

Reply:
I am not sure what you mean by `So that the function def exists(self) can still infer as property.`. The implementation of `exists` in the base class calls the overridden versions of `filename` which are implemented in child classes.

Reply:
you get this typical error with mypy as follows:

``` error: Argument 1 to "endswith" of "str" has incompatible type "Callable[[], str]"; expected "Union[str, Tuple[str, ...]]"```

what I'm trying to say, i'm making `filename` in ComponentReleaseNotes as abstract property, which is a read-only property.

Reply:
Interesting. I think that combination of `@property` and `@abstractmethod` doesn't work. It looks like a bug: https://stackoverflow.com/questions/68763357/strange-behaviour-when-mixing-abstractmethod-classmethod-and-property-decorator, so just leave it out.

Reply:
I just saw this in the doc @dblock https://docs.python.org/3/library/abc.html#abc.abstractproperty adding `@property` with `@abstractmethod` should give a read-only property.